File: doma-core/src/main/java/org/seasar/doma/jdbc/criteria/declaration/HavingDeclaration.java
package org.seasar.doma.jdbc.criteria.declaration;

import java.util.Objects;
import org.seasar.doma.jdbc.criteria.context.SelectContext;

/** The having declaration. */
public class HavingDeclaration extends ComparisonDeclaration {

  public HavingDeclaration(SelectContext context) {
    super(() -> context.having, h -> context.having = h);
    Objects.requireNonNull(context);
  }
}

File: doma-criteria/src/main/kotlin/org/seasar/doma/criteria/declaration/HavingDeclaration.kt
package org.seasar.doma.criteria.declaration

import org.seasar.doma.criteria.context.Criterion
import org.seasar.doma.jdbc.Config

@Declaration
class HavingDeclaration(
    config: Config,
    add: (Criterion) -> Unit
) : ComparisonDeclaration<HavingDeclaration>(config, add, ::HavingDeclaration), AggregateDeclaration
